irohad/main/application.hpp

```diff
@@ -166,6 +166,24 @@ class Irohad {
       opt_mst_gossip_params_;
 
   // ------------------------| internal dependencies |-------------------------
+ public:
+  shared_model::crypto::Keypair keypair;
+  std::shared_ptr<iroha::ametsuchi::Storage> storage;
+
+ protected:
+  logger::Logger log_;
+
+  // initialization objects
+  iroha::network::OnDemandOrderingInit ordering_init;
+  iroha::consensus::yac::YacInit yac_init;
+  iroha::network::BlockLoaderInit loader_init;
+
+  // common objects factory
+  std::shared_ptr<shared_model::interface::CommonObjectsFactory>
```

Comment:
Technically `common_objects_factory_` should be above `std::shared_ptr<iroha::ametsuchi::Storage> storage`
https://github.com/hyperledger/iroha/blob/d617982653b0cf8b03e349f937fafac37bce37e2/irohad/main/application.cpp#L122-L136

Reply:
I agree but I don't really want to do a protected-public-protected mix of fields. I think it could be done later.